### PR TITLE
[core] add `Signal.Handler` class to install shutdown hook

### PR DESCRIPTION
**Pull request by hearnadam**
This is step 1 of solving #491. Next, we need a hook to shutdown the scheduler/interrupt tasks.

### CODE REVIEW

**Comment by hearnadam** [comment]
The JVM reflection code is based on ZIO's implementation by @kyri-petrou https://github.com/zio/zio/blob/series/2.x/core/jvm/src/main/scala/zio/internal/Signal.scala

I took the liberty of using `Result` to properly route errors to the log message.

**Comment by kyri-petrou** [comment]
As a side note, as this approach relies on java reflection, it might be worth adding these files to make the signal handler compatible with graalvm's native-image compilation

**Comment by hearnadam** [comment]
Yes, I noticed that. My plan is to tackle all our configuration for reflection at once, since I think we use reflection elsewhere.

File: kyo-core/jvm/src/main/scala/kyo/internal/SignalPlatformSpecific.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private val logger = Logger.getLogger("kyo.internal.Signal")
+    val handle: Handler = {
+        for
+            signalClass        <- findClass("sun.misc.Signal")
+            signalHandlerClass <- findClass("sun.misc.SignalHandler")
+            lookup = MethodHandles.lookup()
+            constructorHandle  <- initSignalConstructorMethodHandle(lookup, signalClass)
+            staticMethodHandle <- initHandleStaticMethodHandle(lookup, signalClass, signalHandlerClass)
+        yield SunMisc(signalHandlerClass, constructorHandle, staticMethodHandle)
+    }.fold { error =>
```
Comment (fwbrasil) [inline kyo-core/jvm/src/main/scala/kyo/internal/SignalPlatformSpecific.scala:26]: It seems `Abort.recover` wrapping the `for` would be cleaner

**Review by fwbrasil — APPROVED** [review]

**Pull request merged**